cloud-sql/mysql/Program.cs

```diff
@@ -65,6 +73,142 @@ private static void ReadAppSettings()
 
     static class ProgramExtensions
     {
+        public static DbConnection OpenWithRetry(this DbConnection connection)
+        {
+            // [START cloud_sql_mysql_dotnet_ado_backoff]
+            connection = Policy
+                .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                .WaitAndRetry(new[]
+                {
+                TimeSpan.FromSeconds(1),
+                TimeSpan.FromSeconds(2),
+                TimeSpan.FromSeconds(5)
+                }, (result, timeSpan, retryCount, context) =>
+                {
+                    // Log any warnings here.
+                })
+                .Execute(() =>
+                {
+                    // Return a new connection.
+                    try {
+                        connection.Open();
+                    }
+                    catch (MySqlException e)
+                    {
+                         Console.WriteLine(
+                             $"Error connecting to database: {e.Message}");
+                    }
+                    return connection;
+                });
+            // [END cloud_sql_mysql_dotnet_ado_backoff]
+            return connection;
+        }
+
+        public static void InitializeDatabase(this DbConnection connection)
+        {
+            using(connection.OpenWithRetry())
+            {
+                using (var createTableCommand = connection.CreateCommand())
+                {
+                    createTableCommand.CommandText = @"
+                        CREATE TABLE IF NOT EXISTS
+                        votes(
+                            vote_id SERIAL NOT NULL,
+                            time_cast timestamp NOT NULL,
+                            candidate CHAR(6) NOT NULL,
+                            PRIMARY KEY (vote_id)
+                        )";
+                    createTableCommand.ExecuteNonQuery();
+                }
+            }
+        }
+
+        public static MySqlConnectionStringBuilder GetMySqlConnectionString(this DbConnection connection)
+        {
+            MySqlConnectionStringBuilder connectionString; 
+            if (Environment.GetEnvironmentVariable("DB_HOST") != null)
+            {
+                connectionString = NewMysqlTCPConnectionString();
+            }
+            else
+            {
+                connectionString = NewMysqlUnixSocketConnectionString();
+            }
+            // [START cloud_sql_mysql_dotnet_ado_limit]

```

Comment:
Are these recommended values or just some values to demonstrate how to configure the pool?
Either way, a comment describing the nature of the values would be nice, something like:

"The values set here are for demonstration purposes only and you should take care in finding the ones that work best for your application". or
"These are the recommended values because of ...."

Reply:
Added a comment as suggested.